bundle/bundle.go

```diff
@@ -121,6 +121,15 @@ func TryLoad(ctx context.Context) (*Bundle, error) {
 	return Load(ctx, root)
 }
 
+func (b *Bundle) InitializeWorkspaceClient() error {
```

Comment:
You could also reuse it in `WorkspaceClient ` call as well

Reply:
I thought about it, that would make the code worse IMO since this does not reduce the number of lines of code and instead introduces an additional level of abstraction.

Reply:
What is the additional level of abstraction?

The initialization code in `WorkspaceClient` __will__ be executed again, because of `b.clientOnce`. If you instead make this function return the client instance, then it can be captured in `b.client` by `WorkspaceClient` and this function doesn't touch it.

Reply:
I meant that `b.Workspace.Client()` is still called from `b.WorkspaceClient`. The additional level of abstraction here refers to `InitializeWorkspaceClient` itself.

However this also consolidates all client initialization so is probably a good idea. Done.